src/theory/theory_model.h

```diff
@@ -130,12 +130,26 @@ class TheoryModel : public Model
   void printRepresentativeDebug( const char* c, Node r );
   /** print representative function */
   void printRepresentative( std::ostream& out, Node r );
+private:
+  /** uf models */
+  std::map< Node, Node > d_uf_models;
 public:
   /** whether function models are enabled */
   bool d_enableFuncModels;
-  //necessary information for function models
+  /** a map from functions f to a list of all APPLY_UF terms with operator f */
   std::map< Node, std::vector< Node > > d_uf_terms;
```

Comment:
Can you use unordered maps here and below?

Reply:
These should probably stay as std::map for now since I iterate over them for finite model finding.